adrs/013-if-then-else.md

```diff
@@ -0,0 +1,111 @@
+# ADR 013 - If, Then and Else Syntax
+### Status: Accepted
+### Josh Holbrook
+
+## Context
+
+In order to implement conditional branching, we need to specify a syntax
+for if, then and else. We would like this syntax to match the expectations of
+a period-appropriate BASIC, but we would also like it to support modern idioms,
+particularly multi-line branches.
+
+MSX BASIC supports the following forms:
+
+- `IF <condition> THEN <lineNo | instructions>`
+- `IF <condition> THEN <lineNo | instructions> ELSE <lineNo | instructions>`
+- `IF <condition> GOTO <lineNo>`
+- `IF <condition> GOTO <lineNo> ELSE <lineNo | instructions>`
+
+When supplied a lineNo, BASIC will jump to that line. Otherwise, it will run
+the listed instructions. Note that `Instructions` is what MSX BASIC calls the
+equivalents to Matanuska's "commands", and are likewise colon-separated.
+
+Notable is that MSX BASIC does *not* support multi-line if statements.
+For an example of a BASIC which *does* support them, we can look to BBC BASIC.
+BBC BASIC supports the following:
+
+```
+IF <condition> THEN
+  ...
+ELSE
+  ...
+ENDIF
+```
+
+Also notable is that neither MSX BASIC nor BBC BASIC has an `else if` - though,
+it likely parses `if <cond> then <ins> else if <cond> then <ins> else <ins>` as
+`If(cond, ins, If(cond, ins, ins))`, where the second `if` is simply another
+instruction.
+
+In the case of single-line if statements, we can parse them as single commands,
+something 
+
+## Decision
+
+We will support two kinds of "if"s: single-line instructions (called `ShortIf`
+in the AST) and multi-line collections of instructions (made up of the
+instructions `If`, `ElseIf`, `Else` and `EndIf`. Single-line if instructions
+may be evaluated as commands.
+
+Single-line if statements will initially support the following forms:
+
+- `if <condition> then <instructions> endif`
+- `if <condition> then <instructions> else <instructions> endif`
+
+In these forms, `instructions` may not contain if/else/endif instructions
+used in a multi-line context.
+
+Multi-line if blocks will support forms such as the following:
+
+```
+<line_no> if <condition> then
+  <lines>
+<line_now> else if <condition> then
+  <lines>
+<line_no> else
+  <lines>
+<line_no> endif
+```
+
+along with variants allowing `then` on the following line:
+
+```
+<line_no> if <condition>
+<line_no>   then <instructions>
+  <lines>
+<line_now> else if <condition>
+<line_no>    then <instructions>
+  <lines>
+<line_no> else
+  <lines>
+<line_no> endif
+```
+
+### What's Not Supported
+
+Note that these forms don't support line numbers for `goto`. That decision
+will be made in the future, when `goto` is existing functionality.
+
+Also not supported are an unterminated short if, as in MSX BASIC:
```

Comment:
BBC BASIC supports this form too, to the exclusion of the second form. This decision within BBC BASIC is probably worth calling out.